helix-term/src/handlers/diagnostics.rs

```diff
@@ -21,4 +166,12 @@ pub(super) fn register_hooks(_handlers: &Handlers) {
         }
         Ok(())
     });
+
+    let tx = handlers.pull_diagnostics.clone();
+    register_hook!(move |event: &mut DocumentDidChange<'_>| {
+        if event.doc.config.load().lsp.auto_signature_help {
+            send_blocking(&tx, PullDiagnosticsEvent::Trigger);
```

Comment:
Should this have it's own config option? Seems odd to tie it to signature help

Reply:
I think this is a copy/paste error. Instead we can check if the `event.doc` has a language server that supports pull diagnostics so that we don't even trigger the hook if the doc won't trigger pull diagnostics.

I'm thinking adding a method to Document:

```rust
pub fn has_language_server_with_feature(&self, feature: LanguageServerFeature) -> bool {
    self.language_servers_with_feature(feature).next().is_some()
}
```

and checking that on `event.doc` instead

Reply:
Yes this was a copy/paste error. 

I added @the-mikedavis suggestion.